File: code/2171. Challenge The Zip Case 4-004-challenge-the-zip-case/4-004-challenge-the-zip-case/pre-recording/end/src/main/kotlin/SupportCode.kt
fun exampleOf(description: String, action: () -> Unit) {
  println("\n--- Example of: $description ---")
  action()
}

const val episodeI = "The Phantom Menace"
const val episodeII = "Attack of the Clones"
const val theCloneWars = "The Clone Wars"
const val episodeIII = "Revenge of the Sith"
const val solo = "Solo: A Star Wars Story"
const val rogueOne = "Rogue One: A Star Wars Story"
const val episodeIV = "A New Hope"
const val episodeV = "The Empire Strikes Back"
const val episodeVI = "Return of the Jedi"
const val episodeVII = "The Force Awakens"
const val episodeVIII = "The Last Jedi"
const val episodeIX = "Episode IX"

const val luke = "Luke Skywalker"
const val hanSolo = "Han Solo"
const val leia = "Princess Leia"
const val chewbacca = "Chewbacca"

const val lightsaber = "Lightsaber"
const val dl44 = "DL-44 Blaster"
const val defender = "Defender Sporting Blaster"
const val bowcaster = "Bowcaster"

fun stringFrom(minutes: Int): String {
  val hours = minutes / 60
  val min = minutes % 60
  return String.format("%d:%02d", hours, min)
}

val runtimes = mapOf(
    episodeI to 136,
    episodeII to 142,
    theCloneWars to 98,
    episodeIII to 140,
    solo to 142,
    rogueOne to 142,
    episodeIV to 121,
    episodeV to 124,
    episodeVI to 134,
    episodeVII to 136,
    episodeVIII to 152
)

File: code/1739. Combining Operators Part 2 RxSwift-4-03/RxSwift-4-03/27b-combining-operators-final/RxSwiftPlayground/RxSwiftPlayground.playground/Sources/SupportCode.swift
import Foundation

public func example(of description: String, action: () -> Void) {
  print("\n--- Example of:", description, "---")
  action()
}

public let episodeI = "The Phantom Menace"
public let episodeII = "Attack of the Clones"
public let theCloneWars = "The Clone Wars"
public let episodeIII = "Revenge of the Sith"
public let solo = "Solo"
public let rogueOne = "Rogue One"
public let episodeIV = "A New Hope"
public let episodeV = "The Empire Strikes Back"
public let episodeVI = "Return Of The Jedi"
public let episodeVII = "The Force Awakens"
public let episodeVIII = "The Last Jedi"
public let episodeIX = "Episode IX"

public let luke = "Luke Skywalker"
public let hanSolo = "Han Solo"
public let leia = "Princess Leia"
public let chewbacca = "Chewbacca"

public let lightsaber = "Lightsaber"
public let dl44 = "DL-44 Blaster"
public let defender = "Defender Sporting Blaster"
public let bowcaster = "Bowcaster"

let formatter = DateComponentsFormatter()

public func stringFrom(_ minutes: Int) -> String {
  let interval = TimeInterval(minutes)
  return formatter.string(from: interval) ?? ""
}

public let runtimes = [
  episodeI: 136,
  episodeII: 142,
  theCloneWars: 98,
  episodeIII: 140,
  solo: 142,
  rogueOne: 142,
  episodeIV: 121,
  episodeV: 124,
  episodeVI: 134,
  episodeVII: 136,
  episodeVIII: 152
]
